### PR TITLE
circle stroke tessellation

### LINKED ISSUE #40: Implement circle stroke tessellation.
A function `stroke_circle` should be added in [basic_shapes.rs](https://github.com/nical/lyon/blob/master/tessellation/src/basic_shapes.rs). The interface would be similar to `fill_circle`, using a `GeometryBuilder<StrokeVertex>` for the output.

The other tessellation routines in the same file provide a good set of examples for how to tessellate specific shapes with the `GeometryBuilder` API.

### PR DESCRIPTION
**Pull request by nivkner**
fixes #40

### CODE REVIEW

**Comment by nivkner** [comment]
Just a friendly reminder for @nical to review this PR.

File: tessellation/src/basic_shapes.rs [hunk 1]
```diff
@@ -560,6 +560,31 @@ fn fill_border_radius<Output: GeometryBuilder<FillVertex>>(
...
+	let angle_size = (angle.0 - angle.1).abs();
+	let starting_angle = angle.0.min(angle.1);
+	let closed = angle.0 % (2.0 * PI) == angle.1 % (2.0 * PI);
+
+	let points = (0..resolution)
+		.map(|i| {
+			let new_angle = i as f32 * (angle_size) / resolution as f32 + starting_angle;
+			let normal = vec2(new_angle.cos(), new_angle.sin());
+			center + normal * radius
+			});
```
Comment (nical) [inline tessellation/src/basic_shapes.rs]: Another formatting nit:
```rust
    let points = (0..resolution).map(|i| {
        let new_angle = i as f32 * (angle_size) / resolution as f32 + starting_angle;
        let normal = vec2(new_angle.cos(), new_angle.sin());
        center + normal * radius
    });
```

File: tessellation/src/basic_shapes.rs [hunk 2]
```diff
@@ -560,6 +560,31 @@ fn fill_border_radius<Output: GeometryBuilder<FillVertex>>(
...
 
+// tessellate the stroke of rounded corners.
+fn stroke_border_radius<Output: GeometryBuilder<StrokeVertex>>(
+    center: Point,
+    angle: (f32, f32),
+    radius: f32,
+    resolution: u32,
+    output: &mut Output
+) {
+	if resolution == 0 {
```
Comment (nical) [inline tessellation/src/basic_shapes.rs]: nit: indentation is slightly off.

**Comment by nivkner** [comment]
Fixed formatting as requested

File: tessellation/src/basic_shapes.rs [hunk 3]
```diff
@@ -637,6 +664,36 @@ pub fn fill_circle<Output: GeometryBuilder<FillVertex>>(
     return output.end_geometry();
 }
 
+/// Tessellate the stroke of a circle.
+pub fn stroke_circle<Output>(center: Point, radius: f32, tolerance: f32, output: &mut Output) -> Count
+    where Output:
+GeometryBuilder<StrokeVertex>
```
Comment (nical) [inline tessellation/src/basic_shapes.rs]:
```rust
    where Output: GeometryBuilder<StrokeVertex>
```

File: tessellation/src/basic_shapes.rs [hunk 4]
```diff
@@ -560,6 +560,33 @@ fn fill_border_radius<Output: GeometryBuilder<FillVertex>>(
...
+    if resolution == 0 {
+        return;
+    }
+
+    let angle_size = (angle.0 - angle.1).abs();
+    let starting_angle = angle.0.min(angle.1);
+    let closed = angle.0 % (2.0 * PI) == angle.1 % (2.0 * PI);
+
+    let points =
+(0..resolution).map(|i| {
```
Comment (nical) [inline tessellation/src/basic_shapes.rs]:
```rust
    let points = (0..resolution).map(|i| {
```

**Comment by nivkner** [comment]
Forgot github's mobile version does line wrapping. Fixed

**Pull request merged**